forem/forem
# Fix intermittently broken reaction images on home feed

## What type of PR is this? (check all applicable)


## Description

There is a bug where the reaction images on the logged-in home feed are temporarily broken (goes away on refresh, there are also a bunch of Honeybadger error reports). It can be reliably reproduced using keyboard navigation, though it's possibly triggerable in other ways too.

- The reaction count component (`ReactionCount`) and initializers (in `buildArticleHTML`, which gets called by our infinite feed scroll handler) rely on [this little partial](https://github.com/forem/forem/blob/b812ca5955c0d21671c147a2f2639a5663a705a6/app/views/articles/_reaction_category_resources.html.erb) that renders the correct reaction icons into the page in a hidden div, which the frontend can then pluck them from
- However, when this bug is triggered the component and initializers are unable to find the hidden div, even though it _should_ exist
- This is almost certainly down to InstantClick and something in its dark innards messing with JS execution when it pre-fetches pages and swaps out the main body of the current page with the main body of the pre-fetched page
- Instead of detangling all that I chose the much smoother road of moving the partial away InstantClick's shenanigans entirely to always be rendered in the footer

## Related Tickets & Documents

- Related Issue #19655
- Closes #19655

## QA Instructions, Screenshots, Recordings

- Visit the home feed while logged in
- Click on any article link (also works with notifications, the about page, etc)
- Go back using the keyboard (cmd + left arrow on mac, alt + left arrow on windows/linux)
- Confirm that the reaction images are not broken

also, to see the bug you can perform the same steps on (current) DEV

### UI accessibility checklist

N/A (only moving a hidden partial around)

## Added/updated tests?

- [x] Yes, fixed up the `ReactionsCount` spec a bit. I couldn't write an automated test for the keyboard navigation because Cypress keyboard events are _janky_.

## [optional] Are there any post deployment tasks we need to perform?

I don't know if/how this will affect caching?
